### ISSUE TITLE
Update of RPM package overwrites existing configuration files

### ISSUE THREAD

**Issue by rafaelorafaelo** [openbao/openbao#599]
**Describe the bug**
Update of RPM package overwrites configuration files in /etc/openbao.  

**To Reproduce**
1. Install OpenBao 2.0.1 `dnf install ./bao_2.0.1_linux_amd64.rpm`  
2. Edit `/etc/openbao/openbao.env` and `/etc/openbao/openbao.hcl`
3. Update OpenBao with `dnf up ./bao_2.0.2_linux_amd64.rpm`
4. Check configuration files, if your changes are still there.

**Expected behavior**
Existing configuration files should not be overwritten. Instead, files like `/etc/openbao/openbao.env.rpmnew` and `/etc/openbao/openbao.hcl.rpmnew` should be created. https://access.redhat.com/solutions/60263

**Environment:**
* OpenBao Server Version (retrieve with `bao status`): 2.0.2
* OpenBao CLI Version (retrieve with `bao version`): v2.0.2
* Server Operating System/Architecture: Rocky Linux 9.4 x86_64

**Comment by ajaygk95** [comment]
I can help with this. I can create a PR for this.

Question: How do we build rpms for openbao in order to test this?
` make bootstrap && make dev` ?

**Comment by cipherboy** [comment]
@ajaygk95 No, we build RPMs via goreleaser: https://goreleaser.com/quick-start/

You'll have to do the transform in our build pipeline first: https://github.com/openbao/openbao/blob/0ce0c4f8f5c645bb694bf3b9301bc62f146f877f/.github/workflows/release.yml#L128

I think it'd be great to have a makefile target for this, if you're willing!

**Comment by ajaygk95** [comment]
I'm new to using gorelease, I tried to execute the go-release command locally and I'm facing the below issues,

```
# sed "s/REPLACE_WITH_RELEASE_GOOS/linux/g" .goreleaser-template.yaml > .goreleaser.yaml
# sed -i "s/^#LINUXONLY#//g" .goreleaser.yaml
# goreleaser release --clean --timeout=60m --verbose --parallelism 2
  • verbose output enabled
  • only configurations files on  version: 2  are supported, yours is  version: 0 , please update your configuration
  • using configuration                              path=.goreleaser.yaml
  • parallelism: 2
  • cleaning distribution directory
  • loading environment variables
  ⨯ release failed after 0s                  error=missing GITHUB_TOKEN, GITLAB_TOKEN and GITEA_TOKEN
```

After setting the GITHUB_TOKEN, I am getting a different error,

```
# goreleaser release --clean --timeout=60m --verbose --parallelism 2
  • verbose output enabled
  • only configurations files on  version: 2  are supported, yours is  version: 0 , please update your configuration
  • using configuration                              path=.goreleaser.yaml
  • parallelism: 2
  • cleaning distribution directory
  • loading environment variables
    • using token from  $GITHUB_TOKEN 
    • token type: github
  • getting and validating git state
    • git command result                             args=[-c log.showSignature=false rev-parse --is-inside-work-tree] stdout=true stderr=
    • git command result                             args=[-c log.showSignature=false rev-parse --is-inside-work-tree] stdout=true stderr=
    • git command result                             args=[-c log.showSignature=false rev-parse --abbrev-ref HEAD --quiet] stdout=main stderr=
    • git command result                             args=[-c log.showSignature=false show --format=%h HEAD --quiet] stdout=0ce0c4f8f5 stderr=
    • git command result                             args=[-c log.showSignature=false show --format=%H HEAD --quiet] stdout=0ce0c4f8f5c645bb694bf3b9301bc62f146f877f stderr=
    • git command result                             args=[-c log.showSignature=false rev-list --max-parents=0 HEAD] stdout=6491b6567387b96bc7545c89050d7d13f2b81548
3f73a1ce9a4529f228e772024716723bc6f4c23b
7f44e0706e651ae49865ea1c004e88b4d75dfbfd
e8521e5ddf9c8fa44437f8f599a83e3b071f6462
2bdec4ee952728afea2aff6f330ae29b55a57433
9f3e8a384f08373560a2bd0ee2cc8c16aac90dee
76b00b55fd1efcfa42c3d60f3ef518e7982f2835
2de267a93ce142b884caa04508106f9e6217e626
ed3069f01d4ee023d296212a3521d3354f109194 stderr=
    • git command result                             args=[-c log.showSignature=false show --format='%ct' HEAD --quiet] stdout='1729092724' stderr=
    • git command result                             args=[-c log.showSignature=false describe --always --dirty --tags] stdout=api/auth/approle/v2.0.1-103-g0ce0c4f8f5 stderr=
    • git command result                             args=[-c log.showSignature=false ls-remote --get-url] stdout=https://github.com/openbao/openbao.git stderr=
    • git command result                             args=[-c log.showSignature=false tag --points-at HEAD --sort -version:refname] stdout= stderr=
    • git command result                             args=[-c log.showSignature=false describe --tags --abbrev=0 HEAD] stdout=api/auth/approle/v2.0.1 stderr=
    • git command result                             args=[-c log.showSignature=false tag -l --format='%(contents:subject)' api/auth/approle/v2.0.1] stdout='Fix api, sdk modules with v2.0.1 (#425)' stderr=
    • git command result                             args=[-c log.showSignature=false tag -l --format='%(contents)' api/auth/approle/v2.0.1] stdout='Fix api, sdk modules with v2.0.1 (#425)

* Update protobuf definitions to use v2 format

Signed-off-by: Alexander Scheel <ascheel@gitlab.com>

* Fix api, sdk modules with v2.0.1

This retracts the v2.0.0 module version that we pushed and finishes the
Go module at /v2. In the future, the following module tagging order will
need to be used:

 - api
 - api/auth/*
 - sdk
 - Any external dependencies (go-kms-wrapping, openbao-template, &c)
 - Finally, the main module

In this case, the api/auth/* modules aren't currently buildable until
the v2.0.1 tag is pushed for api.

Signed-off-by: Alexander Scheel <ascheel@gitlab.com>

---------

Signed-off-by: Alexander Scheel <ascheel@gitlab.com>' stderr=
    • git command result                             args=[-c log.showSignature=false tag -l --format='%(contents:body)' api/auth/approle/v2.0.1] stdout='* Update protobuf definitions to use v2 format

Signed-off-by: Alexander Scheel <ascheel@gitlab.com>

* Fix api, sdk modules with v2.0.1

This retracts the v2.0.0 module version that we pushed and finishes the
Go module at /v2. In the future, the following module tagging order will
need to be used:

 - api
 - api/auth/*
 - sdk
 - Any external dependencies (go-kms-wrapping, openbao-template, &c)
 - Finally, the main module

In this case, the api/auth/* modules aren't currently buildable until
the v2.0.1 tag is pushed for api.

Signed-off-by: Alexander Scheel <ascheel@gitlab.com>

---------

Signed-off-by: Alexander Scheel <ascheel@gitlab.com>' stderr=
    • git command result                             args=[-c log.showSignature=false describe --tags --abbrev=0 tags/api/auth/approle/v2.0.1^] stdout=v2.0.0 stderr=
    • git command result                             args=[-c log.showSignature=false rev-list -n1 v2.0.0] stdout=700fe3f27ab1f0ec39ce20c36f6d9d97c9fe6ac3 stderr=
    • git command result                             args=[-c log.showSignature=false tag --points-at 700fe3f27ab1f0ec39ce20c36f6d9d97c9fe6ac3 --sort -version:refname] stdout=v2.0.0 stderr=
    • git command result                             args=[-c log.showSignature=false status --porcelain] stdout= stderr=
    • git state                                      commit=0ce0c4f8f5c645bb694bf3b9301bc62f146f877f branch=main current_tag=api/auth/approle/v2.0.1 previous_tag=v2.0.0 dirty=false
    • git command result                             args=[-c log.showSignature=false status --porcelain] stdout= stderr=
    • git command result                             args=[-c log.showSignature=false describe --exact-match --tags --match api/auth/approle/v2.0.1] stdout= stderr=fatal: no tag exactly matches '0ce0c4f8f5c645bb694bf3b9301bc62f146f877f'
  ⨯ release failed after 0s                  error=git tag api/auth/approle/v2.0.1 was not made against commit 0ce0c4f8f5c645bb694bf3b9301bc62f146f877f
```

**Comment by cipherboy** [comment]
\o hey @ajaygk95  -- I think you need to either use `--snapshot` (to avoid pushing up to GitHub and thus the need for the token) or use `build` mode.  See https://goreleaser.com/customization/snapshots/?h=snapshot

Also, to set tags, I think you can do something like:

```
export GORELEASER_CURRENT_TAG=v2.1.0
export GORELEASER_PREVIOUS_TAG=v2.0.2
goreleaser release --snapshot ...
```

from: https://goreleaser.com/cookbooks/set-a-custom-git-tag/

**Comment by cipherboy** [comment]
I believe this was resolved by #639.
